feat: replace Decord with video_reader-rs

kozoy opened the pull request:
## Motivation

This PR addresses the maintenance and stability risks posed by the deprecated decord library with the following key motivations:
1. Unmaintained Dependency:
    - decord's last commit was 3 years ago, raising concerns about long-term compatibility and security.
    - Known unresolved issues include memory leaks (dmlc/decord#323) and segmentation faults (dmlc/decord#329).
2. Modernization & Stability:
    - Replaces decord's obsolete FFmpeg 4 dependency with video_reader-rs, which offers:
      - Active maintenance (latest release yesterday)
      - Lower memory usage and competitive performance
      - Simplified installation

## Modifications

Modification
1. Dependency Replacement
    - Replaced decord with [video_reader-rs](https://github.com/gcanat/video_reader-rs) for video processing
    - Updated dependency specifications and installation docs
2. Configuration Improvements
    - constant var configured via server_args
3. Add missing image_sizes field
4. Removes redundant codes

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [x] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [x] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [x] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [x] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [x] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

kozoy committed:
feat: replace Decord

zhyncs commented:
Great work!

fecet commented:
should link to https://github.com/sgl-project/sglang/issues/5116

merrymercy commented on python/sglang/lang/backend/runtime_endpoint.py:
```diff
@@ -36,7 +36,8 @@ def __init__(
         self.api_key = api_key
         self.verify = verify
-
+        # 如果这里export http_proxy的话
```
use english

KAOZUOI replied:
sorry, fixed

merrymercy commented:
can you resolve the conflicts?

KAOZUOI commented:
> can you resolve the conflicts?

KAOZUOI commented:
> can you resolve the conflicts?

conflicts resolved

fzyzcjy commented on python/sglang/utils.py:
```diff
@@ -127,9 +127,21 @@ def http_request(
         headers["Authorization"] = f"Bearer {api_key}"
 
+    if is_local:
+        if os.getenv("http_proxy") or os.getenv("https_proxy"):
```
nit: to make the PR quickly merged, it would be great to keep the PR minimal i.e. only do bare minimal change related to video_reader-rs. And in a second separate PR the features like http proxy or port selection can be implemented.

(btw for proxy, we may need to consider HTTP_PROXY, HTTPS_PROXY etc)

AlienKevin commented:
@mickqian I noticed that #5888 was merged several days ago, could this this PR also be merged? Thanks.

kozoy committed:
feat: faster video loader

kozoy committed:
feat: faster video loader

JustinTong0323 committed:
fix lint

Signed-off-by: Xinyuan Tong <justinning0323@outlook.com>

JustinTong0323 committed:
Fix: video FPS retrieval

Updates the method used to retrieve video frames per second (FPS)
to ensure accurate video processing. Also fixes the way to read frame from video.

Signed-off-by: Xinyuan Tong <justinning0323@outlook.com>

JustinTong0323 committed:
Merge branch 'main' into base_mmproc

DiweiSun commented:
@JustinTong0323 @mickqian @zhyncs , It seems that this PR has resulted in a CI failure. Could you kindly conduct another review to decide whether a revert is required?
https://github.com/sgl-project/sglang/actions/runs/16305585455/job/46050937843

mickqian commented:
reverted. It seems the wheel of the package is incomplete. Could you contact the author? @kozoy

AlienKevin commented:
@mickqian @kozoy I raised an issue in video_reader-rs: https://github.com/gcanat/video_reader-rs/issues/67
